<launch>
    <arg name="local_robot_id" default="0" />
    <arg name="other_robot_id" default="0" />

    <arg name="topic_left_image_rect" value="/robot_$(arg local_robot_id)/camera/infra1/image_rect_raw"/>
    <arg name="topic_right_image_rect" value="/robot_$(arg local_robot_id)/camera/infra2/image_rect_raw"/>
    <arg name="topic_left_camera_info" value="/robot_$(arg local_robot_id)/camera/infra1/camera_info"/>
    <arg name="topic_right_camera_info" value="/robot_$(arg local_robot_id)/camera/infra2/camera_info"/>
    <arg name="topic_rgb_image_rect" value="/robot_$(arg local_robot_id)/camera/color/image_raw" />

    <arg name="log_pos" default="true" />
    <arg name="logs_location" default="/root/rdpgo_ws/src/robust_distributed_slam_module/scripts/log/"/>
    
    <arg name="fcu_port" default="/dev/ttyTHS2" />
    <arg name="baud_rate" default="115200" />

    <include file="$(find multi_robot_separators)/launch/multi_robot_separators.launch">
    <!-- all vars that included.launch requires must be set -->
        <arg name="other_robot_id" value="$(arg other_robot_id)"/>
        <arg name="local_robot_id" value="$(arg local_robot_id)"/>
        <arg name="frame_id" value="/robot_$(arg local_robot_id)/camera_link"/>
        <arg name="estimate_stereo_transform_from_tf" value="true" />

        <arg name="topic_left_image_rect" value="$(arg topic_left_image_rect)"/>
        <arg name="topic_right_image_rect" value="$(arg topic_right_image_rect)"/>
        <arg name="topic_left_camera_info" value="$(arg topic_left_camera_info)"/>
        <arg name="topic_right_camera_info" value="$(arg topic_right_camera_info)"/>
        <arg name="topic_rgb_image_rect" value="$(arg topic_rgb_image_rect)" />

        <arg name="use_estimates_of_poses" value="true" />

        <arg name="log_pos" value="$(arg log_pos)" />
        <arg name="logs_location" value="$(arg logs_location)" />
        <arg name="fcu_port" value="$(arg fcu_port)" />
        <arg name="baud_rate" value="$(arg baud_rate)" />
    </include>
</launch>
